src/NuGet.Core/NuGet.Packaging/Rules/PathTooLongRule.cs

```diff
@@ -0,0 +1,40 @@
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using NuGet.Common;
+
+namespace NuGet.Packaging.Rules
+{
+    public class PathTooLongRule : IPackageRule
+    {
+        private int _pathLenghtWarningThreshold = 200;
+        public string MessageFormat { get; }
+
+        public PathTooLongRule(string messageFormat)
+        {
+            MessageFormat = messageFormat;
+        }
+
+        public IEnumerable<PackagingLogMessage> Validate(PackageArchiveReader builder)
+        {
+            var packageIdentity = builder.GetIdentity();
+            var versionFolderPathResolver = new VersionFolderPathResolver(string.Empty);
+            var installedPath = versionFolderPathResolver.GetInstallPath(packageIdentity.Id, packageIdentity.Version);
```

Comment:
no ,this give the global path, we don't calculate user packages folder path here.

Reply:
@zhili1208 it will only give you the combined string of "packageId/Version" without any packages folder either user or global since you're passing empty root path to `VersionFolderPathResolver`. I thought this was intentional but if not then look at it again.

Reply:
Yes, this was intentional